macros/utils/cross_db_utils/current_timestamp.sql

```diff
@@ -1,63 +1,20 @@
-{# 
-   Copyright 2021 dbt Labs, Inc.
-
-   Licensed under the Apache License, Version 2.0 (the "License");
-   you may not use this file except in compliance with the License.
-   You may obtain a copy of the License at
-
-       http://www.apache.org/licenses/LICENSE-2.0
-
-   Unless required by applicable law or agreed to in writing, software
-   distributed under the License is distributed on an "AS IS" BASIS,
-   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
-   See the License for the specific language governing permissions and
-   limitations under the License. 
-#}
-
-{# 
-    dbt_utils.current_timestamp is using dbt_utils.type_timestamp which is going to be deprecated from dbt-utils version 0.9.0,
-    therefore dbt_utils.current_timestap is going to be deprecated as well (without having a similar macro at dbt-core).
-    We copied this macro, with a small change that supports the deprecation, so we could continue and use it in our project.
-#}
 {% macro current_timestamp() -%}
-  {{ return(adapter.dispatch('current_timestamp', 'elementary')()) }}
+  {% if dbt_version >= '1.2.0' %}
+        {# This macro is depricated from dbt_utils version 0.9.0, but still hasn't got an equivalent macro at dbt-core #}
+        {# Should be replaced to the equivalent macro once it released #}
+        {{ return(dbt_utils.current_timestamp()) }}
+    {% else %}
+        {{ return(dbt_utils.current_timestamp()) }}
+    {% endif %}
 {%- endmacro %}
 
-{% macro default__current_timestamp() %}
-    current_timestamp::{{elementary.type_timestamp()}}
-{% endmacro %}
-
-{% macro redshift__current_timestamp() %}
-    getdate()
-{% endmacro %}
-
-{% macro bigquery__current_timestamp() %}
-    current_timestamp
-{% endmacro %}
 
-
-{# 
-    dbt_utils.current_timestamp_in_utc is using dbt_utils.type_timestamp and dbt_utils.current_timestamp which are going to be deprecated from dbt-utils version 0.9.0,
-    therefore dbt_utils.current_timestamp_in_utc is going to be deprecated as well (without having a similar macro at dbt-core).
-    We copied this macro, with a small change that supports the deprecation, so we could continue and use it in our project.
-#}
 {% macro current_timestamp_in_utc() -%}
```

Comment:
Why do we need this then?

Reply:
Just as prep for the future?

Reply:
Yes.
We will have to create this "in between" macro and use it instead of dbt-utils directly.
We already replaced the use of them in the previous PR, so this is nice to have an easy preparation for the future 🙂

Reply:
ok, approved